File: user-service/src/main/java/rs/edu/raf/repository/CompanyRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.user.Company;

import java.util.Optional;

@Repository
public interface CompanyRepository extends JpaRepository<Company, Long> {
    Optional<Company> findByTIN(Integer TIN);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/repository/CompanyRepository.kt
package org.example.userservicekotlin.repository

import org.example.userservicekotlin.model.user.Company
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CompanyRepository : JpaRepository<Company, Long> {
    fun findByTIN(TIN: Int): Optional<Company>
}